lib/src/socket.dart

```diff
@@ -443,8 +449,11 @@ class PhoenixSocket {
         err,
         stacktrace,
       );
-      return false;
-    } on WebSocketChannelException catch (err, stacktrace) {
+
+      // Rethrow instead of returning false to ensure the original
+      // PhoenixException is propagated to the call site.
+      rethrow;
```

Comment:
Question: should we `_triggerChannelExceptions` on `PhoenixException` as well? It seems odd only to trigger on other types of exceptions 🤔

Reply:
When `_sendHeartbeat()` is called from `connect()`, if it returns `false` (which is the case when `sendMessage(heartbeat)` throws `PhoenixException`), then it throws `PhoenixException` and catches it right after so that it can schedule a reconnect.

My assumption is that in this case we assume that's something wrong with the socket and try to reconnect, whereas the case of `sendMessage(heartbeat)` throwing `WebSocketChannelException`, then it's a problem with the channel, which calls `_triggerChannelExceptions()` in order to attempt a channel rejoin.

Should we then call `_triggerChannelExceptions()` for exceptions other than `WebSocketChannelException`?

Reply:
Ah fair point. Let me adjust that.

Reply:
I've reverted the code so we only call `_triggerChannelExceptions` on `WebSocketChannelException`. In all other cases, we now `rethrow` to propagate the underlying error to the caller so it can reconnect.